weakmap.c

```diff
@@ -801,6 +908,20 @@ wkmap_delete(VALUE self, VALUE key)
  *    map.getkey(key) -> existing_key or nil
  *
  *  Returns the existing equal key if it exists, otherwise returns +nil+.
+ *
+ *  This might be useful for implementing caches, so that only one copy of
+ *  some object would be used everywhere in the program:
+ *
+ *    Money = Data.define(:amount, :currency)
```

Comment:
I appreciate the `Data` was committed by you, but this could be expressed more simply using a hash/array/string/time instead of subclass of `Data`.

Reply:
Yeah, it wasn't to promote `Data` (I swear :joy:), just to demonstrate "small value object," but I agree it can be done with less extra concepts.